### PR TITLE
refactor: make acbs authentication api tests re-usable

### PR DESCRIPTION
**Pull request by lmarrai-sw**
## Introduction
(Nearly) all of the TFS API tests are going to test that it returns a 500 error if the ACBS authentication process fails. 
https://github.com/UK-Export-Finance/tfs-api/pull/11 adds API tests for checking this, but it would be better if they were re-usable.

## Resolution
I've extracted the tests to a function that can be called in any test file to run the tests.
The function has options to customise the tests for the endpoint under test.
It also returns a function that can be used to mock the authentication process to succeed, and the `idToken` that it will succeed with.

It currently relies on the calling code to clean up `nock` correctly in an `afterEach` hook. We could move that into the extracted function. From most use cases that I can imagine, this should be fine. The downside is that there are some edge cases where the calling code wouldn't want these tests to clean up `nock` themselves

### CODE REVIEW

**Comment by lmarrai-sw** [comment]
I've marked this as in-draft because the branch it's based on has not been merged to `main` yet

File: test/common-tests/acbs-authentication-api-tests.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  makeRequest: () => request.Test;
+}
+
+interface AcbsAuthenticationTestHooks {
+  idToken: string;
+  givenAuthenticationWithTheIdpSucceeds: () => void;
+}
+
+// Note that these tests assume the caller will use nock to abort pending requests
+// and clean all mocks after each test
```
Comment (lmarrai-sw) [inline test/common-tests/acbs-authentication-api-tests.ts:17]: This function relies on the calling code to clean up `nock` correctly in an `afterEach` hook. We could move that into the function instead if we'd prefer. From most use cases that I can imagine, it should be fine to move it into this function. The downside is that there are some edge cases where the calling code wouldn't want these tests to clean up `nock` themselves

**Comment by lmarrai-sw** [comment]
Closing as I can open this in the UK-Export-Finance repo now that APIM-76 has been merged
